vm/runtime.ts

```diff
@@ -666,6 +666,9 @@ module J2ME {
     TWO_PWR_32_DBL = 4294967296,
     TWO_PWR_63_DBL = 9223372036854776000,
 
+    // The target amount of free memory before garbage collection is forced on unwind.
+    FREE_MEMORY_TARGET = 4086,
```

Comment:
> We set ASMJS_TOTAL_MEMORY to 32MiB in the default Makefile, and we change its value to 8MiB in runwhatsapppreview.mk, so we should set FREE_MEMORY_TARGET in the Make files too, so we can set it relative to ASMJS_TOTAL_MEMORY.

Erm, this was based on my faulty understanding that FREE_MEMORY_TARGET is being set to 4086K, i.e. about 4MB, whereas @brendandahl corrected me on IRC that it's actually being set to 4086 *bytes*, to minimize the frequency with which we force collection, while also leaving some space for the collector to allocate memory if needed (although we aren't sure if it's actually needed.)

In that case, we don't need to adjust *FREE_MEMORY_TARGET* relative to *ASMJS_TOTAL_MEMORY*. We might want to adjust this value in the future, depending on how effectively it balances the various concerns. But we can do that just easily here as in the Make files. So there's no good reason to move it.